algorithms.tex

```diff
@@ -8,6 +8,143 @@
 
 \subsection{Numerics}
 
+The \texttt{Float} class holds an arbitrary-precision binary floating-point value
+and a precision in bits. An operation between two \texttt{Float}
+inputs is rounded to the larger of the two precisions.
+
+The preferred way to evaluate an expression numerically is with the
+\texttt{evalf} method, which internally estimates the number of accurate
+bits of the floating-point
+approximation for each sub-expression, and adaptively increases the
+working precision until the estimated accuracy of the
+final result matches the sought number of decimal digits.
+
+The internal error tracking does not provide rigorous error bounds
+(in the sense of interval arithmetic) and cannot be used to track
+uncertainty in measurement data in any meaningful way;
+the sole purpose is to mitigate loss of accuracy that typically occurs
+when converting symbolic expressions to numerical values, for example
+due to catastrophic cancellation. This is illustrated by the following
+example:
+
+\begin{verbatim}
+>>> cos(exp(-100)).evalf(25) - 1
+0
+>>> (cos(exp(-100)) - 1).evalf(25)
```

Comment:
You don't state anywhere that `evalf(25)` means to use 25 decimal digits of precision (although it may be obvious).